eras/conway/impl/src/Cardano/Ledger/Conway/Genesis.hs

```diff
@@ -41,26 +45,30 @@ instance Crypto c => DecCBOR (ConwayGenesis c)
 instance Crypto c => EncCBOR (ConwayGenesis c)
 
 instance Crypto c => ToCBOR (ConwayGenesis c) where
-  toCBOR cg@(ConwayGenesis _) =
-    let ConwayGenesis {cgGenDelegs} = cg
-     in toEraCBOR @(ConwayEra c) $
-          encode $
-            Rec ConwayGenesis
-              !> To cgGenDelegs
+  toCBOR ConwayGenesis {..} =
+    toEraCBOR @(ConwayEra c) $
+      encode $
+        Rec ConwayGenesis
+          !> To cgGenDelegs
+          !> To cgUpgradePParams
 
 instance Crypto c => FromCBOR (ConwayGenesis c) where
   fromCBOR =
     eraDecoder @(ConwayEra c) $
       decode $
         RecD ConwayGenesis
           <! From
+          <! From
 
 instance Crypto c => ToJSON (ConwayGenesis c) where
-  toJSON cg@(ConwayGenesis _) =
-    let ConwayGenesis {cgGenDelegs} = cg
-     in object ["genDelegs" .= toJSON cgGenDelegs]
+  toJSON ConwayGenesis {..} =
+    object
+      [ "genDelegs" .= toJSON cgGenDelegs
+      , "upgradePParams" .= toJSON cgUpgradePParams
```

Comment:
This data structure should be mostly flat. See the json instance for alonzo pparams

UpgradePParams should not be exposed to the user, it should be an internal implementation detail